test/unit/spec/default_spec.rb

```diff
@@ -0,0 +1,121 @@
+
+  describe 'action :install' do
+    platform_package_matrix.each do |platform, versions|
+      versions.each do |version, packages|
+        describe "on #{platform} #{version}" do
+          packages.each do |package, expected_url|
+            include_examples 'standard install', platform, version, package, expected_url
+          end
+        end
+      end
+    end
+
+    context 'with a non-package name' do
+      let(:test_params) { { name: 'Logmaster', package: :universal_forwarder, build: 'cae2458f4aef', version: '6.3.4' } }
+
+      it 'should set the run state' do
+        expect(run_state['current_installation']).to eq(
+          name: 'Logmaster',
+          package: :universal_forwarder,
+          version: '6.3.4',
+          build: 'cae2458f4aef',
+          x64: true
+        )
+      end
+
+      it 'should install from the rpm package' do
+        expect(chef_run).to install_rpm_package('splunkforwarder')
+      end
+    end
+
+    context 'with base_url' do
+      let(:expected_url) do
+        'https://repo.internet.website/splunk/universalforwarder/releases/6.3.4/linux/splunkforwarder-6.3.4-cae2458f4aef-linux-2.6-x86_64.rpm'
+      end
+      let(:package_path) { "./test/unit/.cache/#{filename_from_url(expected_url)}" }
+      let(:test_params) do
+        {
+          package: :universal_forwarder,
+          build: 'cae2458f4aef',
+          version: '6.3.4',
+          base_url: 'https://repo.internet.website/splunk'
+        }
+      end
+
+      it 'should download the package' do
+        expect(chef_run).to create_remote_file(package_path).with(source: expected_url)
+      end
+    end
+
+    context 'when package is not specified' do
+      let(:test_params) { { name: 'hotcakes', build: 'cae2458f4aef', version: '6.3.4' } }
+
+      it 'fails to install' do
+        expect { chef_run }.to raise_error(/Package must be specified.*/)
+      end
+    end
+
+    context 'when version is not specified' do
+      let(:test_params) { { build: 'cae2458f4aef' } }
+
+      it 'fails to install' do
+        expect { chef_run }.to raise_error(Chef::Exceptions::ValidationFailed, /.* version is required/)
+      end
+    end
+
+    context 'when build is not specified' do
+      let(:test_params) { { version: '6.3.4' } }
+
+      it 'fails to install' do
+        expect { chef_run }.to raise_error(Chef::Exceptions::ValidationFailed, /.* build is required/)
+      end
+    end
+  end
+
+  describe 'action :uninstall' do
+    platform_package_matrix.each do |platform, versions|
+      versions.each do |version, packages|
+        context "on #{platform} #{version}" do
+          packages.each do |package, _|
+            include_examples 'standard uninstall', platform, version, package
+          end
+        end
+      end
+    end
+
+    context 'when package is not specified' do
+      let(:test_params) { { name: 'hotcakes', action: :uninstall } }
+      it 'fails to install' do
```

Comment:
nit, fails to uninstall.  Or possibly reword to just 'fails the chef run' in which case you could do the same for all the other failure tests so they can all be safely copy pasted around.